cli/core/pkg/command/config.go

```diff
@@ -216,39 +216,29 @@ var initConfigCmd = &cobra.Command{
 		if cfg.ClientOptions.CLI == nil {
 			cfg.ClientOptions.CLI = &configapi.CLIOptions{}
 		}
-		repos := cfg.ClientOptions.CLI.Repositories
-		finalRepos := []configapi.PluginRepository{}
-		for _, repo := range config.DefaultRepositories {
-			var exists bool
-			for _, r := range repos {
-				if repo.GCPPluginRepository.Name == r.GCPPluginRepository.Name {
-					finalRepos = append(finalRepos, r)
-					exists = true
-				}
-			}
-			if !exists {
-				finalRepos = append(finalRepos, repo)
-			}
+
+		serverName := ""
+		server, err := cfg.GetCurrentServer()
+		if err == nil && server != nil {
+			serverName = server.Name
 		}
-		cfg.ClientOptions.CLI.Repositories = finalRepos
 
-		err = configlib.StoreClientConfig(cfg)
+		serverPluginDescriptors, standalonePluginDescriptors, err := pluginmanager.InstalledPlugins(serverName)
 		if err != nil {
 			return err
 		}
-		// TODO: cli.ListPlugins is deprecated: Use pluginmanager.AvailablePluginsFromLocalSource or pluginmanager.AvailablePlugins instead
-		descriptors, err := cli.ListPlugins()
+
+		for _, desc := range append(serverPluginDescriptors, standalonePluginDescriptors...) {
+			config.AddDefaultFeatureFlagsIfMissing(cfg, desc.DefaultFeatureFlags)
```

Comment:
Older plugins can take care of updating this DefaultFeatureFlag by themselves because of the way they are importing the tanzu-framework as a dependency. There is a `func init` that gets run every-time which configures this feature-flags if they are missing with each plugin invocation.

Reply:
ah, thanks.